pkg/capi/core.go

```diff
@@ -0,0 +1,149 @@
+			kind = "AzureMachineTemplate"
+		case configv1.GCPPlatformType:
+			gcpMapiMachineSpec = getGCPMAPIProviderSpec(client)
+			failureDomain = gcpMapiMachineSpec.Zone
+			kind = "GCPMachineTemplate"
+		default:
+			Skip(fmt.Sprintf("Platform %v does not support , skipping.", platform))
+		}
+
+	})
+
+	AfterEach(func() {
+		// if the current testing are skipped, we skip clean resources
+		if CurrentSpecReport().State == gotypes.SpecStateSkipped {
+			return
+		}
+
+		framework.DeleteCAPIMachineSets(ctx, client, machineSet)
+		framework.WaitForCAPIMachineSetsDeleted(ctx, client, machineSet)
+		switch platform {
+		case configv1.AWSPlatformType:
+			framework.DeleteObjects(ctx, client, awsMachineTemplate)
+		case configv1.AzurePlatformType:
+			framework.DeleteObjects(ctx, client, azureMachineTemplate)
+		case configv1.GCPPlatformType:
+			framework.DeleteObjects(ctx, client, gcpMachineTemplate)
+		default:
+			Skip(fmt.Sprintf("Platform %v does not support , skipping.", platform))
+		}
+	})
+
+	// OCP-75779 - [CAPI] Labels and annotations specified in a machineset should propagate to nodes.
+	// author: huliu@redhat.com
+	It("should be able to run a machine with labels and annotatioins and they are propagated to nodes", func() {
+		switch platform {
+		case configv1.AWSPlatformType:
+			awsMachineTemplate = newAWSMachineTemplate(awsMapiMachineSpec)
+			Expect(client.Create(ctx, awsMachineTemplate)).To(Succeed(), "Failed to create awsmachinetemplate")
+		case configv1.AzurePlatformType:
+			azureMachineTemplate = newAzureMachineTemplate(client, azureMapiMachineSpec)
+			Expect(client.Create(ctx, azureMachineTemplate)).To(Succeed(), "Failed to create azuremachinetemplate")
+		case configv1.GCPPlatformType:
+			gcpMachineTemplate = createGCPMachineTemplate(gcpMapiMachineSpec)
+			Expect(client.Create(ctx, gcpMachineTemplate)).To(Succeed(), "Failed to create gcpmachinetemplate")
+			machineTemplateName = gcpMachineTemplate.Name
+		default:
+			Skip(fmt.Sprintf("Platform %v does not support , skipping.", platform))
+		}
+
+		machineSetParams = framework.NewCAPIMachineSetParams(
+			"machineset-75779",
+			clusterName,
+			failureDomain,
+			0,
+			corev1.ObjectReference{
+				Kind:       kind,
+				APIVersion: infraAPIVersion,
+				Name:       machineTemplateName,
+			},
+		)
+		machineSet, err = framework.CreateCAPIMachineSet(ctx, client, machineSetParams)
+		Expect(err).ToNot(HaveOccurred(), "Failed to create CAPI machineset")
+
+		machineSetCopy := machineSet.DeepCopy()
+		machineSetCopy.Spec.Replicas = ptr.To(int32(1))
+		machineSetCopy.Spec.Template.ObjectMeta.Annotations = make(map[string]string)
+		machineSetCopy.Spec.Template.ObjectMeta.Annotations["anno1key"] = "anno1value"
+		machineSetCopy.Spec.Template.ObjectMeta.Labels["label1key"] = "label1value"
+
+		err = client.Patch(ctx, machineSetCopy, runtimeclient.MergeFrom(machineSet))
+		Expect(err).NotTo(HaveOccurred(), "Failed to patch machineset.c")
```

Comment:
should we use `machineset.Name` in place of `machineset.c` here , to know which machineset failed to be patched ?